IntegrationTests/TextInputTest.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+const React = require('react');
+const ReactNative = require('react-native');
+const {View, TextInput} = ReactNative;
+const {TestModule} = ReactNative.NativeModules;
+
+function expectTrue(condition: boolean, message: string): boolean {
+  if (!condition) {
+    throw new Error(message);
+  }
+
+  return condition;
+}
+
+class TextInputTest extends React.Component<{}, $FlowFixMeState> {
+  state = {
+    text: '123',
+  };
+
+  componentDidMount() {
+    if (!TestModule.verifySnapshot) {
+      throw new Error('TestModule.verifySnapshot not defined.');
+    }
+
+    this.updateText();
+  }
+
+  updateText() {
+    this.setState({text: '0'.concat(this.state.text)}, () => {
```

Comment:
Improvement: Set the natively the cursor position at the beginning of the input field.